homeassistant/components/forecast_solar/__init__.py

```diff
@@ -0,0 +1,77 @@
+"""The Forecast.Solar integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from forecast_solar import ForecastSolar
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import (
+    CONF_AZIMUTH,
+    CONF_DAMPING,
+    CONF_DECLINATION,
+    CONF_MODULES_POWER,
+    DOMAIN,
+)
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Forecast.Solar from a config entry."""
+    api_key = entry.options.get(CONF_API_KEY)
+    if not api_key:
+        api_key = None
+
+    forecast = ForecastSolar(
+        api_key=api_key,
+        latitude=entry.data[CONF_LATITUDE],
+        longitude=entry.data[CONF_LONGITUDE],
+        declination=entry.options[CONF_DECLINATION],
+        azimuth=(entry.options[CONF_AZIMUTH] - 180),
+        kwp=(entry.options[CONF_MODULES_POWER] / 1000),
+        damping=entry.options.get(CONF_DAMPING, 0),
+    )
+
+    # Free account have a resolution of 1 hour, using that as the default
+    # update interval. Using a higher value for accounts with an API key.
+    update_interval = timedelta(hours=1)
+    if api_key is not None:
+        update_interval = timedelta(minutes=15)
```

Comment:
A personal account has a resolution of 30 minutes. Can we derive from the API key or an API call what type of account it is? Or maybe it's a header in the response or we can derive from response when we call again?

Reply:
That is not part of the info available, unfortunately. Hence it picks to be on the safe side for either case.

You think we should use different values? I wanted to be on the safe side for the remote service as well.

Reply:
Thinking about it, I guess we could...
We get more or less info, so let set it from 15 -> to 30 to match the personal account. I believe @klaasnicolaas wanted to add more sensors based on what is available on your account, as soon as that is in, we can determine the right interval based on that.